hashicorp/dev-portal
# feat: custom consul-dot-io docs landing

## "Ready for Review" checklist




---

## Relevant links

- [Preview link][preview] 🔎
- [Asana task](https://app.asana.com/0/1201987349274776/1202080053968887/f) 🎟️

## What

Builds a new custom `/docs` landing page for Consul.io.

## Why

To improve the user experience of the consul.io/docs landing page.

## How

Takes an identical approach to Vault in #178:

- Have split out `docs/[...page].tsx`, which is very similar to the former `[[...page]].tsx`
- New file `docs/index.tsx`, where the custom landing page is built

## Testing

To visit the preview product docs landing page, such as Consul's:

1. Open [the preview link][preview]
2. Select the desired product in the lower-left products dropdown. This switched the site's configuration to serve the product's dot-io site, rather than dev-portal.
3. Navigate to `/docs`

- [ ] Visit the Consul docs landing page, and ensure the page looks and functions as expected
    - The new cards content should appear on the page
    - The sidebar should function as expected
    - Accessing versioned docs should function as expected. Note that the cards content is shown on the latest version only.
- [ ] Visit the Vault docs landing page, and ensure everything continues to render as expected
    - "Under the hood" modifications have been made to this view; previously it was slightly less abstracted since it was only in use by Vault

## Anything else?

Not at the moment!

[preview]: https://dev-portal-git-zscustom-consul-io-docs-landing-hashicorp.vercel.app/docs
